rbac/providers/ldap/ldap_message_validator.py

```diff
@@ -52,11 +49,80 @@ def validate_next_payload(payload):
                 required_field_data, required_field_dn
             )
         )
+
     if not data_node[required_field_dn]:
         raise LdapValidationException(
             "'{0}'.'{1}' cannot be empty".format(required_field_data, required_field_dn)
         )
 
+    if required_field_user_name not in data_node:
+        raise LdapValidationException(
+            "'{0}' is missing an entry for: '{1}'".format(
+                required_field_data, required_field_user_name
+            )
+        )
+
+    if not data_node[required_field_user_name]:
+        raise LdapValidationException(
+            "'{0}'.'{1}' cannot be empty".format(
+                required_field_data, required_field_user_name
+            )
+        )
+
+    if required_field_cn not in data_node:
+        raise LdapValidationException(
+            "'{0}' is missing an entry for: '{1}'".format(
+                required_field_data, required_field_cn
+            )
+        )
+
+    if not data_node[required_field_cn]:
+        raise LdapValidationException(
+            "'{0}'.'{1}' cannot be empty".format(required_field_data, required_field_cn)
+        )
+
+    if required_field_given_name not in data_node:
```

Comment:
In every case where the error is the same (ie: "'{0}' is missing an entry for: '{1}'", etc), the required fields can be grouped and iterated. Something like required_fields_not_empty = [rq_fld_foo, rq_fld_bar....]. It is a small change we can go through when you are in the office.

Reply:
Sure. Sounds good